File: ProjectEuler/P6.py
#!/usr/bin/env python
"""
The sum of the squares of the first ten natural numbers is,
1**2 + 2**2 + ... + 10**2 = 385

The square of the sum of the first ten natural numbers is,
(1 + 2 + ... + 10)**2 = 55**2 = 3025

Hence the difference between the sum of the squares of the first ten natural numbers and the square of the sum is 3025 385 = 2640.

Find the difference between the sum of the squares of the first one hundred natural numbers and the square of the sum.
"""

n = 100
l = range(1,n+1)
l2 = [li*li for li in l]
sum_sq = sum(l2)
sq_sum = sum(l)**2
print sq_sum,sum_sq,sq_sum-sum_sq
    

File: ProjectEuler/P6.cc
#include <iostream>
#include <vector>
#include <numeric>
using namespace std;

int square(int i){return i*i;}

main(){
  int nmax=100;
  vector<int> num(nmax),numsq(nmax);
  for (int i=1; i<=nmax; i++){
    num.push_back(i);
    numsq.push_back(i*i);
  }
  int sq_sum = square(accumulate(num.begin(),num.end(),0));
  int sum_sq = accumulate(numsq.begin(),numsq.end(),0);
  cout << sq_sum <<" "<< sum_sq <<" "<< sq_sum-sum_sq << endl;
}
